proxylib/redis/redcon.go

```diff
@@ -0,0 +1,213 @@
+package redis
+
+// The code below is almoste entirely borrowed from https://github.com/tidwall/redcon
+// commit 56b016127ae730019c5e3073a413ad9526440c38
+//
+// It was not possible to simply use this code as a 3rd-party vendor
+// package, as the parser was tightly-coupled with the underlying I/O layer
+
+import (
+	"errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	errUnbalancedQuotes       = &errProtocol{"unbalanced quotes in request"}
+	errInvalidBulkLength      = &errProtocol{"invalid bulk length"}
+	errInvalidMultiBulkLength = &errProtocol{"invalid multibulk length"}
+	errDetached               = errors.New("detached")
+	errIncompleteCommand      = errors.New("incomplete command")
+	errTooMuchData            = errors.New("too much data")
+)
+
+type errProtocol struct {
+	msg string
+}
+
+func (err *errProtocol) Error() string {
+	return "Protocol error: " + err.msg
+}
+
+// Command represent a command
+type Command struct {
+	// Raw is a encoded RESP message.
+	Raw []byte
+	// Args is a series of arguments that make up the command.
+	Args [][]byte
+}
+
+func parseInt(b []byte) (int, bool) {
+	if len(b) == 1 && b[0] >= '0' && b[0] <= '9' {
+		return int(b[0] - '0'), true
+	}
+	var n int
+	var sign bool
+	var i int
+	if len(b) > 0 && b[0] == '-' {
+		sign = true
+		i++
+	}
+	for ; i < len(b); i++ {
+		if b[i] < '0' || b[i] > '9' {
+			return 0, false
+		}
+		n = n*10 + int(b[i]-'0')
```

Comment:
Did you take into account for a `b` that has more than 10 digits? (2147483647) Can't this overflow?

Reply:
Go int is 64 bits in 64-bit systems, but it still can overflow.